ffi/capi/src/zoned_time_formatter.rs

```diff
@@ -0,0 +1,500 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// @generated by tools/make/codegen/src/capi_datetime.rs
+
+#[rustfmt::skip]
+#[diplomat::bridge]
+#[diplomat::abi_rename = "icu4x_{0}_mv1"]
+#[diplomat::attr(auto, namespace = "icu4x")]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use writeable::TryWriteable;
+
+    #[allow(unused_imports)]
+    use crate::{
+        date_formatter::ffi::{DateFormatter, DateFormatterGregorian},
+        datetime_formatter::ffi::DateTimeLength,
+        date::ffi::IsoDate,
+        datetime_helpers,
+        datetime_helpers::map_or_default,
+        datetime_options::ffi::{DateTimeAlignment, TimePrecision},
+        errors::ffi::DateTimeFormatterLoadError,
+        errors::ffi::DateTimeWriteError,
+        locale_core::ffi::Locale,
+        time_formatter::ffi::TimeFormatter,
+        time::ffi::Time,
+        timezone::ffi::TimeZoneInfo,
+    };
+
+    #[cfg(feature = "buffer_provider")]
+    use crate::provider::ffi::DataProvider;
+
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::datetime::FixedCalendarDateTimeFormatter, Typedef)]
+    pub struct ZonedTimeFormatter(
+        pub  icu_datetime::FixedCalendarDateTimeFormatter<
+            (),
+            icu_datetime::fieldsets::enums::CompositeFieldSet,
+        >,
+    );
+
+    impl ZonedTimeFormatter {
+        #[diplomat::attr(all(supports = fallible_constructors, supports = named_constructors), named_constructor = "specific_long")]
+        #[cfg(feature = "compiled_data")]
+        pub fn create_specific_long(
+            locale: &Locale,
+            length: Option<DateTimeLength>,
+            time_precision: Option<TimePrecision>,
+            alignment: Option<DateTimeAlignment>,
+        ) -> Result<Box<Self>, DateTimeFormatterLoadError> {
```

Comment:
Thought: It's nice that inlining the options here avoids the extra box/clone, but it is then misleading that `length` is a prominent argument but applies only to the day period, a setting that rarely has any effect.